src/bin/casr-cluster.rs

```diff
@@ -416,10 +416,69 @@ fn dedup(indir: &Path, outdir: Option<PathBuf>) -> error::Result<(usize, usize)>
     Ok((before, after))
 }
 
+/// Merge sub directory to main
+///
+/// # Arguments
+///
+/// * `sdir` - path to sub directory
+///
+/// * `mdir` - path to main directory
+///
+/// # Return value
+///
+/// Number of merged reports
+fn merge_dirs(sdir: &Path, mdir: &Path) -> error::Result<u64> {
+    let dir = fs::read_dir(mdir).with_context(|| {
+        format!(
+            "Error occurred while opening directory with Casr reports. File: {}",
+            mdir.display()
+        )
+    })?;
+
+    let mut mainhash = HashSet::new();
+    for entry in dir.flatten() {
+        if entry.path().extension().is_some() && entry.path().extension().unwrap() == "casrep" {
+            if let Ok(trace) = stacktrace(entry.path().as_path()) {
+                mainhash.insert(trace);
+            } else {
+                // return err?
+            }
+        }
+    }
+
+    let dir = fs::read_dir(sdir).with_context(|| {
+        format!(
+            "Error occurred while opening directory with Casr reports. File: {}",
+            sdir.display()
+        )
+    })?;
+
+    let mut new: u64 = 0;
+    for entry in dir.flatten() {
+        if entry.path().extension().is_some() && entry.path().extension().unwrap() == "casrep" {
+            if let Ok(trace) = stacktrace(entry.path().as_path()) {
+                if mainhash.insert(trace) {
+                    fs::copy(
+                        &entry.path().as_path(),
+                        &Path::new(&mdir).join(entry.file_name()),
+                    )?;
+                    new += 1;
+                }
+            } else {
+                // return err?
```

Comment:
We may just log an error here and continue merging.

Reply:
done